scheduler/src/cook/monitor.clj

```diff
@@ -145,18 +146,30 @@
   (run!
     (fn [[user stats]]
       (run! (fn [[type amount]]
-              (-> [state user (name type) (str "pool-" pool-name)]
+              (do
+                (-> [state user (name type) (str "pool-" pool-name)]
                   counters/counter
-                  (set-counter! amount)))
+                  (set-counter! amount))
+                ; Metrics need to be pre-registered in prometheus, so only record them if the metric exists
+                ; Log a warning otherwise so that we know to add a metric if we add a new resource type.
+                (if (and (contains? prometheus/resource-metric-map state) (contains? (prometheus/resource-metric-map state) type))
+                  (prometheus/set ((prometheus/resource-metric-map state) type) {:pool pool-name :user user} amount)
+                  (log/warn "Encountered unknown keys for prometheus user metrics; state" state "type" type))))
```

Comment:
Is there a way (and/or does it make sense to) push this down a layer so that checks like this  always happen? I see that ln 155 and ln 170 differ in that one has state and type and the other only has state.